spec/factories/omniauth.rb

```diff
@@ -0,0 +1,33 @@
+FactoryGirl.define do
+
+  factory :auth_hash, class: OmniAuth::AuthHash do
+    skip_create
+
+    initialize_with do
+      OmniAuth::AuthHash.new({
+        provider: provider,
+        uid: uid,
+        info: {
+          email: email
+        }
+      })
```

Comment:
Fair enough. I suppose it's reasonable to expect users of a factory to understand how it's been configured before calling it.